File: ui/src/api/registration.ts
import { apiClient } from './client';

export interface MemberInfo {
  id: string;
  name: string;
  mobile: string;
  flatNumber?: string;
  wing?: string;
  role: string;
  designation?: string;
}

export interface InitiateResponse {
  message: string;
  member: MemberInfo;
  otpExpiry: number;
}

export interface VerifyOTPResponse {
  message: string;
  verified: boolean;
}

export interface RegistrationCompleteResponse {
  message: string;
  user: {
    id: string;
    name: string;
    email: string;
    phone: string;
    role: string;
  };
}

export const registrationApi = {
  initiate: async (mobile: string): Promise<InitiateResponse> => {
    return apiClient.post<InitiateResponse>('/registration/initiate', { mobile }, {
      skipAuth: true,
    });
  },

  verifyOTP: async (mobile: string, otp: string): Promise<VerifyOTPResponse> => {
    return apiClient.post<VerifyOTPResponse>('/registration/verify-otp', { mobile, otp }, {
      skipAuth: true,
    });
  },

  complete: async (mobile: string, email: string, password: string): Promise<RegistrationCompleteResponse> => {
    return apiClient.post<RegistrationCompleteResponse>('/registration/complete', {
      mobile,
      email,
      password,
    }, {
      skipAuth: true,
    });
  },

  resendOTP: async (mobile: string): Promise<{ message: string }> => {
    return apiClient.post<{ message: string }>('/registration/resend-otp', { mobile }, {
      skipAuth: true,
    });
  },
};

File: server/internal/dto/response/registration.go
package response

// MemberInfoResponse contains member info for registration
type MemberInfoResponse struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Mobile      string `json:"mobile"`
	FlatNumber  string `json:"flatNumber,omitempty"`
	Wing        string `json:"wing,omitempty"`
	Role        string `json:"role"`
	Designation string `json:"designation,omitempty"`
}

// InitiateRegistrationResponse after OTP is sent
type InitiateRegistrationResponse struct {
	Message    string             `json:"message"`
	Member     MemberInfoResponse `json:"member"`
	OTPExpiry  int                `json:"otpExpiry"` // seconds
}

// VerifyOTPResponse after successful OTP verification
type VerifyOTPResponse struct {
	Message string `json:"message"`
	Verified bool   `json:"verified"`
}

// RegistrationCompleteResponse after successful registration
type RegistrationCompleteResponse struct {
	Message string       `json:"message"`
	User    UserResponse `json:"user"`
}
